[Aloysius Neo Ze En] iP

Aloynz opened the pull request:
### DukeLite

> If you embrace that the things that you can do are limitless, you can put your ding in the universe. 
You can change the world. - Tim Cook [(source)](https://github.com/Aloynz/ip/releases)

DukeLite is a **light weighted** application which helps you to keep track of tasks. It's,
- **simple** to use
- **rich** in features
- **_EASY_** to pick up

All you need to do is,
1.  download it from [here](https://github.com/Aloynz/ip/releases)
2. double click the .jar file to run
3. add your tasks.
4. let it track your tasks 😍

Features :

- [x]  Managing tasks
- [x] Managing deadlines, event
- [ ]  Display a GUI (coming soon)

What's a better way than not using DukeLite?

```ruby
public class Main {

    public static void main(String[] args) {

         System.out.println("DUKELITE IS AWESOME!!!"); 
    }
}
```

Commit:
Add Gradle support

Aloynz committed:
Added Greet Echo Exit Feature

Aloynz committed:
Added Add, List Feature

Aloynz committed:
Added Mark as Done Feature

Aloynz committed:
Added ToDo, Event, Deadline Feature

Aloynz committed:
Added Automated Text UI Testing

Aloynz committed:
Added Exception Handling and updated Automated Testing

Aloynz committed:
Added Delete Feature and updated Automated Testing

Aloynz committed:
Added enums

Aloynz committed:
Add Save function

Aloynz committed:
Add .gitignore

Aloynz committed:
Add Dates and Times

Use Date Objects instead of storing it as strings.

This is to standarise the dates input by the users.

Aloynz committed:
Merge branch 'branch-Level-8'

Aloynz committed:
Use Object Oriented Programming

UI class to deal with interactions with user.
Storage class to deal with loading and saving tasks.
Paser class to deal with user command.
Tasklist class to perform add/delete operations.

Aloynz committed:
Add packages

Aloynz committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

Aloynz committed:
Add JUnit Tests

Test the parser to parse String inputs to Command object.

Test the workings of the commands TODO and DELETE.

Aloynz committed:
Add Javadocs for java classes

Aloynz committed:
Edit code to follow Coding Standard

Removed duplicate code from Storage.java

Ensured that every line has at most 120 characters.

Aloynz committed:
Add find by keyword feature

Tasks which match the the keyword will be displayed.

Aloynz committed:
Improve Coding Standards

Aloynz committed:
Implement find by keyword.

Gives users the list of tasks which contains the keyword in the description.

Aloynz committed:
Use Checkstyle to detect coding violations.

Aloynz committed:
Add GUI

Icons of the bot and user is added.

charleslimjh commented on src/main/java/seedu/duke/Duke.java:
```diff
@@ -0,0 +1,74 @@
+    /*Ui component to handle the user interface of the program */
+    private Ui ui;
+    /*Parser object to parse user input into recognised Commands */
+    private Parser parser;
```
```suggestion
    /** Storage object to handle reading and writing to hard disk */
    private Storage storage;
    /** Tasklist containing an arraylist of tasks */
    private TaskList tasks;
    /** Ui component to handle the user interface of the program */
    private Ui ui;
    /** Parser object to parse user input into recognised Commands */
    private Parser parser;
```
I think Javadocs for class members should still start with `/**` according to the coding standard. Might be good to take note?

charleslimjh commented on src/main/java/seedu/duke/Command.java:
```diff
@@ -0,0 +1,160 @@
+                tasks.add(task);
+                ui.add(task);
+                output += "Now you have " + tasks.size() + (tasks.size() == 1 ? " task" : " tasks")
+                + " in the list.";
```
```suggestion
                output += "Now you have " + tasks.size() 
                        + (tasks.size() == 1 ? " task" : " tasks")
                        + " in the list.";
```

I like the line break to improve readability, but line indentation for wrapped lines should be 8 spaces. Maybe consider this change instead? Same change can be applied in other commands in `Command.java`.

charleslimjh commented on src/main/java/seedu/duke/Deadline.java:
```diff
@@ -0,0 +1,47 @@
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + "(by: " + by.getMonth() + "-" + by.getDayOfMonth() + "-" + by.getYear()
+            + " " + String.format("%02d", by.getHour()) + ":" + String.format("%02d", by.getMinute()) + ")";
```
Not sure I like how many characters are in each line here. While it does not hit the suggested 110-120 character limit, would it be better to break each line further ahead?

charleslimjh commented on src/main/java/seedu/duke/Task.java:
```diff
@@ -0,0 +1,60 @@
+    public String setUndone() {
+        this.isDone = false;
+        return "OK, I've marked this task as not done yet:\n[ ] " + this.description;
+    }
```
Looks like a minor typo in Javadocs, the method description should start with a verb in plural form.
```suggestion
    /**
     * Sets the task to be completed.
     */
    public String setDone() {
        this.isDone = true;
        return "Nice! I've marked this task as done:\n[X] " + this.description;
    }

    /**
     * Sets the task to be uncompleted.
     */
    public String setUndone() {
        this.isDone = false;
        return "OK, I've marked this task as not done yet:\n[ ] " + this.description;
    }
```

charleslimjh reviewed (commented):
Overall, the code looks good and adheres to the coding conventions, save for minor typos here and there! 😄 

In certain areas, there were too many line breaks as well. It might be good to relook at some of your longer functions and remove any unnecessary whitespace to reduce the LoC in your project.

Aloynz committed:
Add Assertions

Assertions are added to check if the tasks stored in the harddisk
are of the right format.

By ensuring the tasks stored in the harddisk have the right
format, the tasks can then be loaded correctly.

Aloynz committed:
Edit mainClassName in build.gradle file

Aloynz committed:
Improve Code Quality

Delete dead code and minimise code duplication. Commented code
which are no longer used are removed and duplicate code is also
removed for better quality in the implementation.

By minimising code duplication, it is easier to reduce the amount
of error prone codes written.

Aloynz committed:
Merge pull request #1 from Aloynz/branch-A-Assertions

Add Assertions

Aloynz committed:
Merge branch 'master' into branch-A-CodeQuality

Aloynz committed:
Merge pull request #2 from Aloynz/branch-A-CodeQuality

Improve code quality

Aloynz committed:
Add tags to tasks

User will be allowed to label a task with a tag with the command
tag {task number} {name of tag}.

This helps the user to better categorise the list of their tasks.

Aloynz committed:
Add colours to the program

Added a colour for the background and the send button.

Aloynz committed:
Add representative screenshot of product and product name on title
bar of window

Aloynz committed:
Add User Guide in README

Aloynz committed:
Update build.gradle

Aloynz committed:
Edit build.gradle

Aloynz committed:
Fix DateTime and GUI bugs

Aloynz committed:
Fix bugs

Aloynz committed:
Add Javadocs

Aloynz committed:
Resolve checkstyle errors

Aloynz committed:
Use SLAP in lengthy classes
